### PR TITLE
Feat/incubator text field preset

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Support `preset` prop in the new TextField component and add the first "default" preset. 
This should keep parity with the old TextField UI but still allows to keep the UI code separated 

## Changelog
Support "default" preset to our Incubator.TextField component

### CODE REVIEW

File: src/incubator/TextField/index.tsx [hunk 1]
```diff
@@ -83,9 +84,13 @@ type TextFieldProps = MarginModifiers &
      * Container style of the whole component
      */
     containerStyle?: ViewStyle;
+    /**
+     * Predefined preset to use for styling the field
+     */
+    preset?: 'default' | undefined;
```
Comment (mendyEdri) [inline src/incubator/TextField/index.tsx:90]: Is the type could be optional boolean? 
(`useDefaultPreset` for example)

**Review by mendyEdri — COMMENTED** [review]
small comment regarding a type :)

File: src/incubator/TextField/index.tsx (hunk 1 above)
Comment (ethanshar) [reply src/incubator/TextField/index.tsx:90]: Actually the idea was to add more presets in the future, so the `preset` prop was meant to be used as an enum. 
For now I just added the `default` preset so it feels redundant..

Comment (mendyEdri) [reply src/incubator/TextField/index.tsx:90]: Yeah that's what I thought it meant to be, merging

**Review by mendyEdri — APPROVED** [review]

**Pull request merged**